src/lib/shim/src/lib.rs

```diff
@@ -225,6 +219,125 @@ mod tls_thread_shmem {
     }
 }
 
+mod global_manager_shmem {
+    use super::*;
+
+    // This is set explicitly, so needs a Mutex.
+    static INITIALIZER: SelfContainedMutex<Option<ShMemBlockSerialized>> =
+        SelfContainedMutex::const_new(None);
+
+    // The actual block is in a `LazyLock`, which is much faster to access.
+    // It uses `INITIALIZER` to do its one-time init.
+    static SHMEM: LazyLock<ShMemBlockAlias<ManagerShmem>> = LazyLock::const_new(|| {
+        let serialized = INITIALIZER.lock().take().unwrap();
+        unsafe { Serializer::global().deserialize(&serialized) }
+    });
+
+    /// # Safety
+    ///
+    /// `blk` must contained a serialized block of type `ManagerShmem`, which
+    /// outlives the process.
+    pub unsafe fn set(blk: &ShMemBlockSerialized) {
+        assert!(INITIALIZER.lock().replace(*blk).is_none());
+        SHMEM.force();
+    }
+
+    /// Panics if `set` hasn't been called yet.
+    pub fn get() -> impl core::ops::Deref<Target = ShMemBlockAlias<'static, ManagerShmem>> + 'static
+    {
+        SHMEM.force()
+    }
+
+    pub fn try_get(
+    ) -> Option<impl core::ops::Deref<Target = ShMemBlockAlias<'static, ManagerShmem>> + 'static>
+    {
+        if !SHMEM.initd() {
+            None
+        } else {
+            Some(get())
+        }
+    }
```

Comment:
This seems kind of confusing to me. I would expect a `try_get` to try to get the object, and return `None` if it couldn't. But `try_get` doesn't even try to get the object. The serialized object may be ready in `INITIALIZER`, but `try_get` won't even try deserializing it. You must call `get` first before `try_get` will ever work. I guess this is relying on the fact that `set()` calls `SHMEM.force()` (which is a `get()`). I think the documentation here could be improved a bit.